lib/core/source/localSource/tips_local_source.dart

```diff
@@ -20,13 +20,28 @@ abstract class TipsLocalSource {
   Future<void> deleteAllTips();
 
   @transaction
-  Future<void> replaceTips(List<TipDbEntity>? tips) async {
-    await deleteAllTips();
-    if (tips != null) {
-      await insertTips(tips);
+  Future<void> replaceAndUpdateTips(List<TipDbEntity> tips) async {
+    final oldTips = await getTips().first;
+    var mapTips = <String, TipDbEntity>{};
+    mapTips = Map.fromIterable(oldTips, key: (e) => e.id);
+
+    final mergedList = tips.map((value) => updateTip(value, mapTips[value.id]));
+
+    if (mergedList.isNotEmpty) {
+      await deleteAllTips();
+      await insertTips(mergedList.toList());
     }
-    if (tips == null || tips.isEmpty) {
+    if (mergedList.isEmpty) {
       changeListener.add('tips');
     }
   }
+
+  TipDbEntity updateTip(TipDbEntity newTip, TipDbEntity? oldTip) {
+    if (oldTip != null) {
+      newTip
+        ..randomId = oldTip.randomId
+        ..favourite = oldTip.favourite;
+    }
+    return newTip;
+  }
```

Comment:
i think we can move it to one line if we do something like this 
```suggestion
  TipDbEntity updateTip(TipDbEntity newTip, TipDbEntity? oldTip) => newTip
        ..randomId = oldTip.randomId ?? newTip.randomId
        ..favourite = oldTip.favourite ?? newTip.favourite;
```